### ISSUE TITLE
Conditionally add blocks

### ISSUE THREAD

**Issue by boxfoot** [andywer/webpack-blocks#242]
As I've started using webpack blocks, I've found it helpful to have a mechanism for conditionally adding blocks to the configuration -- like `env`, but keyed off any conditional, not just `NODE_ENV`.

I wrote the following block - wanted to see if this is something others are doing, and if it makes sense to include in the core:

```js
/**
 * Applies an array of webpack blocks only if the given condition is true
 * @param {boolean} condition         Condition to check
 * @param {Function[]} configSetters  Array of functions as returned by webpack blocks.
 * @return {Function}
 */
function when(condition, configSetters) {
  //assertConfigSetters(configSetters)
	if (condition) {
		return group(configSetters)
	} else {
		return () => (config) => config
	}
}
```

This could also allow reimplementing `env` like this (although it's already so simple this probable doesn't help at all)

```js
function env(envName, configSetters) {
     const currentEnv = process.env.NODE_ENV || 'development'
     return when(envName === currentEnv, configSetters);
}
```

**Comment by andywer** [comment]
Hey @boxfoot! Thanks for sharing.

Having a `when` block seems to provide a nicer syntax than using the ternary operator, `.concat()` and friends. I am just not sure if adding such a thing to the core is a terrible idea or actually pretty handy...

@sapegin Any thoughts on this?

**Comment by sapegin** [comment]
I like it ;-)

**Comment by vlad-zhukov** [comment]
Fixed in #243
